bsp/raspberrypi/rp2040/src/hal/i2c.zig

```diff
@@ -71,22 +214,22 @@ pub const I2C = enum(u1) {
         });
     }
 
-    /// Initialise the I2C HW block.
-    pub fn apply(i2c: I2C, comptime config: Config) u32 {
-        const peri_freq = (comptime config.clock_config.get_frequency(.clk_sys)) orelse @compileError("clk_sys must be set for I²C");
-
-        const regs = i2c.get_regs();
-
+    /// Initializes the I2C HW block per the Config provided
+    /// Per the API limitations discussed above, the following settings are fixed and not configurable:
+    /// - Controller Mode only
+    /// - Fast Mode only (full backwards compatibility in the I2C spec means this is suitable even for standard mode baud rates, IE 100kHz)
+    /// - TX_EMPTY_CTRL is always enabled for easy detection of TX finished
+    /// - TX and RX FIFO detection thresholds set to 1, this makes polling for TX finished/RX ready much simpler
+    /// - DREQ signalling is always enabled, harmless if DMA isn't configured to listen for this
+    pub fn apply(i2c: I2C, comptime config: Config) ConfigError!void {
         i2c.disable();
 
-        // Configure as a fast-mode master with RepStart support, 7-bit addresses
-        regs.IC_CON.write(.{
+        i2c.get_regs().IC_CON.write(.{
```

Comment:
i think we should call `get_regs` only once and store the result in this function, makes it easier to optimize :)

Reply:
Good catch! Agreed :+1: